Index: sip4-4.16.2+dfsg/configure.py
===================================================================
--- sip4-4.16.2+dfsg.orig/configure.py	2014-07-04 11:25:52.463702009 -0400
+++ sip4-4.16.2+dfsg/configure.py	2014-07-04 11:25:52.455702009 -0400
@@ -231,6 +231,8 @@
         ("siplib", "siplib.c"),
         ("siplib", "siplib.sbf")
     )
+    if not os.path.exists(patched[0][0]):
+        os.mkdir(patched[0][0])
 
     # The siplib directory may not exist if we are building away from the
     # source directory.
@@ -240,7 +242,7 @@
         pass
 
     for f in patched:
-        dst_fn = os.path.join(*f)
+        dst_fn = os.path.join(src_dir, *f)
         src_fn = os.path.join(src_dir, dst_fn + ".in")
 
         siputils.inform("Creating %s..." % dst_fn)
