Index: sip4-4.14/configure.py
===================================================================
--- sip4-4.14.orig/configure.py	2012-09-29 07:26:55.000000000 -0400
+++ sip4-4.14/configure.py	2012-10-01 02:04:38.522747304 -0400
@@ -224,6 +224,8 @@
         ("siplib", "siplib.c"),
         ("siplib", "siplib.sbf")
     )
+    if not os.path.exists(patched[0][0]):
+        os.mkdir(patched[0][0])
 
     # The siplib directory may not exist if we are building away from the
     # source directory.
@@ -233,7 +235,7 @@
         pass
 
     for f in patched:
-        dst_fn = os.path.join(*f)
+        dst_fn = os.path.join(src_dir, *f)
         src_fn = os.path.join(src_dir, dst_fn + ".in")
 
         siputils.inform("Creating %s..." % dst_fn)
