Index: sip4-4.14.3/siputils.py
===================================================================
--- sip4-4.14.3.orig/siputils.py	2013-02-13 00:46:54.345833434 -0500
+++ sip4-4.14.3/siputils.py	2013-02-13 00:47:24.253832550 -0500
@@ -1652,9 +1652,12 @@
         mfile.write("\n$(OFILES): $(HFILES)\n")
 
         for mf in self._build["moc_headers"].split():
-            root, discard = os.path.splitext(mf)
+            root, _ = os.path.splitext(mf)
             cpp = "moc_" + root + ".cpp"
 
+            if self._src_dir != self.dir:
+                mf = os.path.join(self._src_dir, mf)
+
             mfile.write("\n%s: %s\n" % (cpp, mf))
             mfile.write("\t$(MOC) -o %s %s\n" % (cpp, mf))
 
