Index: sip4-4.14.7/siputils.py
===================================================================
--- sip4-4.14.7.orig/siputils.py	2013-06-17 20:40:58.468338051 -0400
+++ sip4-4.14.7/siputils.py	2013-06-17 20:40:58.460338051 -0400
@@ -1693,9 +1693,12 @@
         mfile.write("\n$(OFILES): $(HFILES)\n")
 
         for mf in self._build["moc_headers"].split():
-            root, discard = os.path.splitext(mf)
+            root, _ = os.path.splitext(mf)
             cpp = "moc_" + root + ".cpp"
 
+            if self._src_dir != self.dir:
+                mf = os.path.join(self._src_dir, mf)
+
             mfile.write("\n%s: %s\n" % (cpp, mf))
             mfile.write("\t$(MOC) -o %s %s\n" % (cpp, mf))
 
