Index: sip4-4.12.4/siputils.py
===================================================================
--- sip4-4.12.4.orig/siputils.py	2011-08-11 16:16:42.116674115 -0400
+++ sip4-4.12.4/siputils.py	2011-08-11 16:16:47.892674112 -0400
@@ -1583,9 +1583,12 @@
         mfile.write("\n$(OFILES): $(HFILES)\n")
 
         for mf in self._build["moc_headers"].split():
-            root, discard = os.path.splitext(mf)
+            root, _ = os.path.splitext(mf)
             cpp = "moc_" + root + ".cpp"
 
+            if self._src_dir != self.dir:
+                mf = os.path.join(self._src_dir, mf)
+
             mfile.write("\n%s: %s\n" % (cpp, mf))
             mfile.write("\t$(MOC) -o %s %s\n" % (cpp, mf))
 
